src/syscalls/stub.rs

```diff
@@ -0,0 +1,257 @@
+//! This module implements syscalls in old ckb-std style based on
+//! an impl of SyscallImpls trait.
+
+use crate::{
+    ckb_constants::{CellField, HeaderField, InputField, Source},
+    error::SysError,
+    syscalls::traits::SyscallImpls,
+};
+use alloc::{boxed::Box, string::String, vec::Vec};
+use core::ffi::CStr;
+
+pub use crate::syscalls::internal::SpawnArgs;
+
+static mut IMPLS: Option<Box<dyn SyscallImpls>> = None;
+
+/// Initializes a new SyscallImpls trait impl.
+pub fn init(impls: Box<dyn SyscallImpls>) {
+    unsafe { IMPLS = Some(impls) }
+}
+
+/// # Safety
+///
+/// This shall be safe since CKB-VM uses a single threaded environment.
+fn get() -> &'static Box<dyn SyscallImpls> {
+    // https://doc.rust-lang.org/nightly/edition-guide/rust-2024/static-mut-references.html#no_std-one-time-initialization
+    let Some(impls) = (unsafe { &*&raw const IMPLS }) else {
+        panic!("No IMPLS provided!");
+    };
+    impls
+}
```

Comment:
Maybe we can try using OnceCell to avoid unsafe operations.

```rs
use core::cell::OnceCell;

static IMPLS: OnceCell<Box<dyn SyscallImpls>> = OnceCell::new();

pub fn init(impls: Box<dyn SyscallImpls>) {
    IMPLS.set(impls).expect("IMPLS has already been initialized");
}
```

Reply:
I've added a comment in code explaining why `OnceCell` does not suit the needs.